narwhals/expr.py

```diff
@@ -466,9 +389,10 @@ def any(self: Self) -> Self:
         """
         return self.__class__(
             lambda plx: self._to_compliant_expr(plx).any(),
-            is_order_dependent=self._is_order_dependent,
-            changes_length=False,
-            aggregates=True,
+            ExprMetadata(
+                kind=ExprKind.AGGREGATION,
+                is_order_dependent=self._metadata["is_order_dependent"],
```

Comment:
Is this expected? Doesn't this aggregation set `is_order_dependent=False`? Similar for other aggregations

Reply:
if we do `nw.col('a').cum_sum().mean()`, then the entire expression is order-dependent (as it depends on the order in which you evaluate the cumulative sum)